### PR TITLE
Fix #1899: rename `hhasher` related structs

### LINKED ISSUE #1899: Rename hhasher related structs
* [x] HHasher
* [x] EndorsementHashMap/Set
* [x] AddressHashMap/Set
* [x] BlockHashMap/Set
* [x] OperationHashMap/Set

### PR DESCRIPTION
**Pull request by yvan-sraka**
The core purpose of this PR is to simplify a lot codebase by using transparent and explicit types rather than alias defined in models: an `AddressHashSet` became an `Set<Address>` and so on!

### CODE REVIEW

File: massa-protocol-worker/src/protocol_controller.rs [hunk 1]
```diff
@@ -92,189 +92,17 @@ pub async fn start_protocol_controller(
     ))
 }
 
-//put in a module to block private access from Protocol_worker.
-mod nodeinfo {
```
Comment (yvan-sraka) [inline massa-protocol-worker/src/protocol_controller.rs:96]: Working on this PR, I spotted that this module is just dead code duplicated in `massa-protocol-exports` ... :smiling_face_with_tear:

Comment (damip) [reply massa-protocol-worker/src/protocol_controller.rs:96]: Feel free to remove dead code :)
Why doesn't the compiler warn about it tho ?

**Review by AureliaDolo — APPROVED** [review]

File: massa-models/src/hash.rs [hunk 2]
```diff
@@ -1,41 +1,48 @@
-// Copyright (c) 2021 MASSA LABS <info@massa.net>
```
Comment (damip) [inline massa-models/src/hash.rs:1]: we need to keep this for now

Comment (damip) [reply massa-models/src/hash.rs:1]: also hash.rs is a confusing name. It's not a file dealing with hashes

Comment (damip) [reply massa-models/src/hash.rs:1]: maybe prehashed.rs or something

File: massa-models/src/hash.rs [hunk 3]
```diff
@@ -1,41 +1,48 @@
...
-
+use massa_signature::{PublicKey, Signature};
+use serde::{Deserialize, Serialize};
 use std::collections::{HashMap, HashSet};
-use std::convert::TryInto;
 use std::hash::{BuildHasherDefault, Hasher};
 use std::marker::PhantomData;
 
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PubkeySig {
```
Comment (damip) [inline massa-models/src/hash.rs:10]: I would not have put PubkeySig in this file...

File: massa-models/src/hash.rs (hunk 2 above)
Comment (yvan-sraka) [reply massa-models/src/hash.rs:1]: Fixed :heavy_check_mark:

File: massa-protocol-worker/src/protocol_controller.rs (hunk 1 above)
Comment (adrien-zinger) [reply massa-protocol-worker/src/protocol_controller.rs:96]: :fearful:

Comment (yvan-sraka) [reply massa-protocol-worker/src/protocol_controller.rs:96]: My bad, it was the whole file that was duplicated and not included in the project ...

**Review by adrien-zinger — DISMISSED** [review]

**Review by damip — CHANGES_REQUESTED** [review]

File: massa-models/src/hash.rs (hunk 3 above)
Comment (yvan-sraka) [reply massa-models/src/hash.rs:10]: In a `pubkey.rs` file in `massa-models`?

Comment (damip) [reply massa-models/src/hash.rs:10]: yes,  or in crypto, or in composite, as you prefer. All seem better-fit than prehashed.rs

Comment (yvan-sraka) [reply massa-models/src/hash.rs]: Fixed :heavy_check_mark:

**Review by damip — APPROVED** [review]

**Comment by damip** [comment]
Personnally I would have named them "PreHashSet" and "PreHashMap" instead of "Set" and "Map" which are quite misleading

**Comment by yvan-sraka** [comment]
bors merge

**Comment by yvan-sraka** [comment]
> @yvan-sraka Personally I would have named them "PreHashSet" and "PreHashMap" instead of "Set" and "Map" which are quite misleading

I see it as an abstraction: I split the underlying mechanism (that could be seen as an optimization, hidden in the library) and the user wanted behavior (how the data structure should behave)!

This separation of concern should allow us in future to implement, change or test optimizations, always with respects to the contract made with user (having data structure that still behaves like a `Set` or a `Map`), that how I see all this!

**Pull request merged**